crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -211,7 +211,22 @@ where
         }
         let block_hash =
             self.client().block_hash_for_id(at)?.ok_or_else(|| EthApiError::UnknownBlockNumber)?;
-        let (cfg, env) = self.cache().get_evm_env(block_hash).await?;
+        let next_base_fee = self.
+            block_by_id(at)
+            .await?
+            .ok_or_else(|| EthApiError::UnknownBlockNumber)?
+            .header
+            .next_block_base_fee()
+            .unwrap();
+        let (cfg, mut env) = self.cache().get_evm_env(block_hash).await?;
+        let next_time_stamp = self.
+            block_by_id(at)
+            .await?
+            .ok_or_else(|| EthApiError::UnknownBlockNumber)?
+            .header
+            .timestamp + 12;
+        env.basefee = U256::from(next_base_fee);
+        env.timestamp = U256::from(next_time_stamp);
```

Comment:
this is the wrong location.

we're now messing with the env for all calls that are not pending.
we need to do this in the else branch of `if let Some(pending) = self.client().pending_header()?`
because we only want to modify this there

Reply:
Okay nice i pushed some changes that fix this up. Let me know how it looks or if it still needs some work and i'll look into it.